radixdlt-core/radixdlt/src/main/java/com/radixdlt/middleware2/network/ConsensusEventMessage.java

```diff
@@ -90,45 +92,37 @@
 	@DsonOutput(Output.ALL)
 	private final Vote vote;
 
-	ConsensusEventMessage() {
-		// Serializer only
-		this.proposal = null;
-		this.vote = null;
-	}
+	public ConsensusEventMessage(
+		@JsonProperty("proposal") Proposal proposal,
+		@JsonProperty("vote") Vote vote
+	) {
+		if (proposal == null && vote == null) {
+			throw new IllegalStateException("No vote nor proposal are provided for ConsensusEventMessage");
+		}
 
-	ConsensusEventMessage(Proposal proposal) {
-		this.proposal = proposal;
-		this.vote = null;
-	}
+		if (proposal != null && vote != null) {
+			throw new IllegalStateException("Both, vote and proposal are provided for ConsensusEventMessage");
+		}
 
-	ConsensusEventMessage(Vote vote) {
-		this.proposal = null;
+		this.proposal = proposal;
 		this.vote = vote;
 	}
 
-	public ConsensusEvent getConsensusMessage() {
-		ConsensusEvent event = consensusMessageInternal();
-		if (event == null) {
-			throw new IllegalStateException("No consensus message.");
-		}
-		return event;
+	public ConsensusEventMessage(Proposal proposal) {
+		this(proposal, null);
 	}
 
-	private ConsensusEvent consensusMessageInternal() {
-		if (this.proposal != null) {
-			return this.proposal;
-		}
-
-		if (this.vote != null) {
-			return this.vote;
-		}
+	public ConsensusEventMessage(Vote vote) {
+		this(null, vote);
+	}
 
-		return null;
+	public ConsensusEvent getConsensusMessage() {
+		return requireNonNull(coalesce(proposal, vote));
```

Comment:
I'd rather not remove the `if else` statements. It's wrong but we shouldn't hide the antipattern and resolve the issue in the correct way.